packages/react-router/modules/Core.js

```diff
@@ -1,18 +1,18 @@
 import React, { PropTypes } from 'react'
 import pathToRegexp from 'path-to-regexp'
 
-const patternCache = { true: {}, false: {} }
+const patternCache = { truetrue: {}, truefalse: {}, falsefalse: {}, falsetrue: {} }
```

Comment:
This is fine for now, but put a pin here to refactor this relative nonsensical code soon. We're bound to sprout a bug here eventually 🙃